### PR TITLE
Added Offline Caching in the Manga Page

### PR DESCRIPTION
**Pull request by yash-k9**
Offline Cache added to the Manga. This functions same as the offline Anime cache

### CODE REVIEW

File: domain/src/main/java/divyansh/tech/animeclassroom/Manga/MangaHomeDefaultRepo.kt [hunk 1]
```diff
@@ -1,22 +1,48 @@
...
 class MangaHomeDefaultRepo @Inject constructor(
     private val remoteRepo: MangaHomeRemoteRepo,
+    private val localRepo: MangaHomeLocalRepo
 ){
 
     suspend fun getHomePageData(url: String): Flow<ResultWrapper<*>> {
-        //TODO: Add local data first. (caching purpose)
         return flow {
+            val localData = localRepo.getAllManga()
+
```
Comment (justdvnsh) [inline domain/src/main/java/divyansh/tech/animeclassroom/Manga/MangaHomeDefaultRepo.kt:24]: HI @yash-k9 please change this function to the one written in Home page caching.

**Review by justdvnsh — CHANGES_REQUESTED** [review]
@yash-k9 Please make the required changes. Thank you for the contributions.

File: domain/src/main/java/divyansh/tech/animeclassroom/Manga/MangaHomeDefaultRepo.kt (hunk 1 above)
Comment (yash-k9) [reply domain/src/main/java/divyansh/tech/animeclassroom/Manga/MangaHomeDefaultRepo.kt:24]: Hi @justdvnsh, should it be changed to the MangaHomeLocalRepo ?

Comment (justdvnsh) [reply domain/src/main/java/divyansh/tech/animeclassroom/Manga/MangaHomeDefaultRepo.kt:24]: @yash-k9 .
No, in this code, the data would be coming from the database. We need to fetch the data from the internet first, update the database and then show the database items to the user. Caching is only done, when there is no internet, then the user would not be shown an empty screen. Instead, they would still be having some data to see.

Comment (yash-k9) [reply domain/src/main/java/divyansh/tech/animeclassroom/Manga/MangaHomeDefaultRepo.kt:24]: @justdvnsh, I've changed the function, the data flow would be in this order 
Fetch Remote -> Update/Store cache Data -> Emit Stored Data

Comment (yash-k9) [reply domain/src/main/java/divyansh/tech/animeclassroom/Manga/MangaHomeDefaultRepo.kt:24]: Hi @justdvnsh, Are there any changes that are required

**Review by justdvnsh — APPROVED** [review]
Hi @yash-k9 Yeah. These were the changes. Thanks for the update and contributions.

**Comment by yash-k9** [comment]
Thanks @justdvnsh, can you please add the hacktoberfest-accepted label to the PR

**Comment by justdvnsh** [comment]
@yash-k9 Oh, thanks for reminding. Forgot about that

**Pull request merged**